### PR TITLE
Move subsystem-templates/nest.xml to the feature pack

### PR DESCRIPTION
**Pull request by ppalaga**

### CODE REVIEW

**Comment by ppalaga** [comment]
@jpkrohling could you please review and eventually merge?

As the next step I'd try to do the same thing in Accounts, unless you started already?

**Comment by jpkrohling** [comment]
Is this going to Alpha 11 ?

File: hawkular-nest/hawkular-nest-feature-pack/src/main/xsl/configuration/standalone/hawkular-nest-template.xsl [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  <xsl:output method="xml" version="1.0" encoding="UTF-8" indent="yes" xalan:indent-amount="4" standalone="no" />
+  <xsl:strip-space elements="*" />
+
+  <!-- Add the Nest subsystem -->
+  <xsl:template match="/*[name(.)='server']/*[name(.)='extensions']">
+    <xsl:copy>
+      <xsl:apply-templates select="node()|@*"/>
+    </xsl:copy>
+    <system-properties>
```
Comment (jpkrohling) [inline hawkular-nest/hawkular-nest-feature-pack/src/main/xsl/configuration/standalone/hawkular-nest-template.xsl]: If there are properties defined in the "parent", those are then replaced, right?

File: hawkular-nest/hawkular-nest-feature-pack/pom.xml [hunk 2]
```diff
@@ -96,6 +96,100 @@
...
+                <transformationSet>
+                  <dir>${project.build.directory}/feature-pack-resources-tmp</dir>
+                  <stylesheet>${basedir}/src/main/xsl/configuration/standalone/hawkular-nest-subsystems.xsl</stylesheet>
+                  <includes>
+                    <include>configuration/standalone/subsystems.xml</include>
+                  </includes>
+                  <outputDir>${project.build.directory}/feature-pack-resources</outputDir>
+                  <fileMappers>
+                    <fileMapper implementation="org.codehaus.plexus.components.io.filemappers.RegExpFileMapper">
+                      <pattern>^configuration/standalone/(.*).xml$</pattern>
```
Comment (jpkrohling) [inline hawkular-nest/hawkular-nest-feature-pack/pom.xml]: We only support standalone, right? Not `standalone-ha`, or `standalone-full-ha`. If that's still true, then this looks OK.

File: hawkular-nest/hawkular-nest-feature-pack/pom.xml [hunk 3]
```diff
@@ -96,6 +96,100 @@
...
+                  <includes>configuration/standalone/template.xml,configuration/standalone/subsystems.xml</includes>
+                  <outputDirectory>${project.build.directory}/feature-pack-resources-tmp</outputDirectory>
+                </artifactItem>
+              </artifactItems>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+
+      <!-- Generate the standalone*.xml stubs from the upstream originals using XSL transformations -->
```
Comment (jpkrohling) [inline hawkular-nest/hawkular-nest-feature-pack/pom.xml]: This comment isn't completely true, I suppose. 

From what I could understand, each transformation set uses resources matching the pattern as input for the XSL, spitting out the resulting XML into feature-pack-resources. 

One transformation set is for subsystems.xml , the other is for template.xml.

**Comment by jpkrohling** [comment]
Sometimes I wonder if it wouldn't be better to have a "descriptor" in a saner language somewhere outside of the `pom.xml`...

Anyway: review completed.

**Comment by ppalaga** [comment]
> Is this going to Alpha 11 ?

Yes, I'd like it to be the case

File: hawkular-nest/hawkular-nest-feature-pack/src/main/xsl/configuration/standalone/hawkular-nest-template.xsl (hunk 1 above)
Comment (ppalaga) [inline hawkular-nest/hawkular-nest-feature-pack/src/main/xsl/configuration/standalone/hawkular-nest-template.xsl]: > If there are properties defined in the "parent", those are then replaced, right?

Good question. Rather not replaced, but the result will contain <system-properties> twice. 

Do you think I should be more defensive in this specific XSL to cover also the case that the input already contains <system-properties> ?

File: hawkular-nest/hawkular-nest-feature-pack/pom.xml (hunk 2 above)
Comment (ppalaga) [inline hawkular-nest/hawkular-nest-feature-pack/pom.xml]: Also a good question. Nobody asked for other than plain `standalone` yet. Once the requirement comes, we know how to add those other files.

File: hawkular-nest/hawkular-nest-feature-pack/pom.xml (hunk 3 above)
Comment (ppalaga) [inline hawkular-nest/hawkular-nest-feature-pack/pom.xml]: > One transformation set is for subsystems.xml , the other is for template.xml.

Strictly speaking yes, but taken a little bit figuratively, both subsystems.xml  and template.xml are some kind of a stub used for building standalone*.xml files, are they not?

File: hawkular-nest/hawkular-nest-feature-pack/src/main/xsl/configuration/standalone/hawkular-nest-template.xsl (hunk 1 above)
Comment (jpkrohling) [inline hawkular-nest/hawkular-nest-feature-pack/src/main/xsl/configuration/standalone/hawkular-nest-template.xsl]: No, as long as we can remember about it if/once we get this problem :)

File: hawkular-nest/hawkular-nest-feature-pack/pom.xml (hunk 2 above)
Comment (jpkrohling) [inline hawkular-nest/hawkular-nest-feature-pack/pom.xml]: +1

File: hawkular-nest/hawkular-nest-feature-pack/src/main/xsl/configuration/standalone/hawkular-nest-template.xsl (hunk 1 above)
Comment (ppalaga) [inline hawkular-nest/hawkular-nest-feature-pack/src/main/xsl/configuration/standalone/hawkular-nest-template.xsl]: Yes, I think the same :)

File: hawkular-nest/hawkular-nest-feature-pack/pom.xml (hunk 3 above)
Comment (jpkrohling) [inline hawkular-nest/hawkular-nest-feature-pack/pom.xml]: They are, but I found it confusing at first and wondered if there was a `standalone.xml` there. If you want to keep the reference to XML files used to assemble a `standalone.xml` , then I'd reference `standalone/*.xml` instead.

**Comment by jpkrohling** [comment]
I guess it should be OK to have it on Alpha 11. If there's a problem on this part, we'll see it very soon :)

File: hawkular-nest/hawkular-nest-feature-pack/pom.xml (hunk 3 above)
Comment (ppalaga) [inline hawkular-nest/hawkular-nest-feature-pack/pom.xml]: OK, changed to `Enhance various *.xml files taken from our dependencies using XSL transformations`

Comment (jpkrohling) [inline hawkular-nest/hawkular-nest-feature-pack/pom.xml]: +1

**Comment by jpkrohling** [comment]
If Travis is happy, I'm happy as well.

**Comment by ppalaga** [comment]
@jpkrohling thanks!

**Pull request merged**